src/OpenTelemetry.Extensions.Enrichment/EnrichmentServicesExtensions.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="EnrichmentServicesExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Linq;
+using OpenTelemetry.Extensions.Enrichment;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Trace;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static class EnrichmentServicesExtensions
+{
+    public static IServiceCollection AddTraceEnricher<T>(this IServiceCollection services)
```

Comment:
nit: Add XML comments for public API. We should call out that ordering is of significant importance when using this API. Meaning the processor really ought to be added before exporter(s) 😄

Reply:
Added XML comments without mentioning the ordering, though. I use InMemoryExporter in unit tests and some tests register enricher before the exporter, some - after.

The ordering does not matter, because TraceEnrichmentProcessor is added through ConfigureOpenTelemetryTracerProvider which is executed when DI container is built already (= TracerProviderBuilder is built too). Correct me if I am wrong somewhere.

Reply:
Order will matter ya. Processors are executed in the order they are added. So if users does AddGenevaExporter() and then AddEnrichment() because GenevaExporter writes out the data without any kind of buffering, it won't see the enrichment data. Whether or not it will work really depends on the exporter implementation. As far as using TracerProviderBuilder or ConfigureOpenTelemetryTracerProvider APIs to do the "add" they both preserve the ordering. The order of the calls will match the order of the processors in the final pipeline.

Reply:
Added remarks. Thank you